source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -18,20 +19,25 @@ function isChecked(file: HTMLElement): boolean {
 }
 
 function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
-	if (!event.shiftKey || !previousFile) {
+	if (!event.shiftKey) {
+		return;
+	}
+
+	const isFirstToggle = previousFile === undefined;
+
+	previousFile ??= select('.js-file'); // #5484
+	if (!previousFile?.isConnected) {
```

Comment:
Why `isConnect`? This is impossible. `batchToggle` is called when the user clicks inside a `.js-file`, so there's no chance that it doesn't exist or that is not connected.